jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ContainerParameters.java

```diff
@@ -200,11 +209,15 @@ public void setVolumes(List<String> volumes) {
       return ConfigurationPropertyValidator.parseMapProperty(
           System.getProperty(PropertyNames.CONTAINER_LABELS));
     }
-    return labels;
+    return labels.get();
   }
 
   public void setLabels(Map<String, String> labels) {
-    this.labels = labels;
+    this.labels.set(labels);
+  }
+
+  public void setLabels(Provider<Map<String, String>> labels) {
+    this.labels.set(labels);
```

Comment:
We should do this in the new way (e.g., like [this](https://github.com/GoogleContainerTools/jib/pull/3034/files#diff-ec1f44085899ade105be7cc3c4febe441db808c45bd067be6827bd19c2f52958R238)), as explained in #3094. That is, no more setters, and `getLabels()` exposes the `MapProperty`. (This assumes the existing usage of directly assigning a map will continue to work, which I think will be the case.)

Reply:
Thank you @chanseokoh, ill check it out

Reply:
I can refactor jib.to.image and jib.to.tags as well to use the new way

Reply:
Thanks for asking! Since we think this new technique isn't 100% backward-compatible, we want to have this change only for a single config for the next release.